### PR TITLE
FIX-GOPROXY

### PR DESCRIPTION
**Pull request by sumitk03**
fix the GOPROXY being searched in GOROOT/go.env file for recent go upgrades.

[go install fails when GOPROXY is empty](https://github.com/golang/go/issues/62274)

https://github.com/golang/go/issues/62274#issuecomment-1692504981

### CODE REVIEW

File: Dockerfile [hunk 1]
```diff
@@ -2,6 +2,12 @@ FROM public.ecr.aws/bitnami/golang
 
 RUN apt update; apt install -y make git curl bash jq awscli
 
+COPY ./go.env /custom_go.env
+
+RUN touch $(go env GOROOT)/go.env &&  \
+    cat /custom_go.env >> $(go env GOROOT)/go.env &&  \
+    rm /custom_go.env
+
```
Comment (shettyh) [inline Dockerfile:10]: Why not change the base image to offical golang image ? That should already have this by default

File: Dockerfile [hunk 2]
```diff
@@ -1,7 +1,13 @@
-FROM public.ecr.aws/bitnami/golang
+FROM golang:1.21.1-alpine3.18
```
Comment (shettyh) [inline Dockerfile:1]: alpine will not work as `apt` is used for dependencies, use default `golang:latest` or with just a version

Comment (sumitk03) [reply Dockerfile:1]: > alpine will not work as apt is used for dependencies (some packages might not present in alpine also)

checking that only

Comment (sumitk03) [reply Dockerfile:1]: found all the dependencies in apk

**Review by shettyh — APPROVED** [review]

**Pull request merged**